### PR TITLE
docs(variability): add dedicated variance doc

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
First stage of our war on variance: Basic documentation in our repo. 

Adds a new document discussing sources of variance and links to additional resources.

### CODE REVIEW

File: docs/variability.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+### Client Hardware Variability
+
+The hardware on which the webpage is loading can greatly impact performance. applied throttling cannot do much to mitigate this issue. Simulated throttling partially mitigates this issue by capping the theoretical execution time of CPU tasks during simulation.
+
+### Client Resource Contention
+
+Other applications running on the same machine while Lighthouse is running can cause contention for CPU, memory, and network resources. Malware, browser extensions, and anti-virus software have particularly strong impacts on web performance. Multi-tenant or parallel server environments can also suffer from these issues. applied throttling is susceptible to this issue. Simulated throttling partially mitigates this issue by replaying network activity on its own and capping CPU execution.
```
Comment (brendankenny) [inline docs/variability.md]: > Multi-tenant or parallel server environments can also suffer from these issues.

is this referring to running the server on the same machine as the testing browser?

File: docs/variability.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+### Page Nondeterminism
+
+Pages can contain logic that is nondeterministic that changes the way a user experiences a page, i.e. an A/B test that changes the layout and assets loaded or a different ad experience based on campaign progress. This is an intentional and irremovable source of variance. If the page changes in a way that hurts performance, Lighthouse should be able to identify this case. The only mitigation here is on the part of the site owner in ensuring that the exact same version of the page is being tested between different runs.
+
+## Strategies for Dealing With Variance
+
+### Isolate External Factors
+
+- Isolate your page from third-party influence as much as possible. It’s never fun to be blamed for someone else's variable failures.
```
Comment (brendankenny) [inline docs/variability.md]: should we add a note somewhere about reducing variability like this has the possible tradeoff of making the numbers less realistic? e.g. If you test without ads you'll be able to test your own code more reliably, but the TTI may no longer resemble what users see on your actual production site

File: docs/variability.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
+# Score Variability
+
+Lighthouse is often a developer's first exposure to web performance measurement. Whether you're new to performance measurement or an old pro, it's always good to refresh ourselves on all the complexity behind measuring websites.
+
+## Summary
+
+Lighthouse performance scores will change due to inherent variability in web and network technologies even if there hasn't been a code change. Run Lighthouse multiple times and beware of variance before drawing conclusions about a performance-impacting change.
```
[the remark this answers is not in the document]
Comment (patrickhulce) [reply docs/variability.md]: done

File: docs/variability.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+## Sources of Variability
+
+Variability in performance measurement is introduced via a number of channels with different levels of impact. Below is a table containing several common sources of metric variability, the typical impact they have on results, and the extent to which different Lighthouse runtimes are able to mitigate their effect.
```
[the remark this answers is not in the document]
Comment (paulirish) [reply docs/variability.md]: can we also link up the throttling.md ?

File: docs/variability.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
```
[the remark this answers is not in the document]
Comment (patrickhulce) [reply docs/variability.md]: moved it after the descriptions, I think it makes more sense there anyway 👍

File: docs/variability.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+### Tier-1 Network Variability
+
+Network interconnects are generally very stable and have minimal impact but cross-geo requests, i.e. measuring performance of a Chinese site from the US, can start to experience a high degree of latency introduced from tier-1 network hops. Applied throttling partially mask these effects with network throttling. Simulated throttling mitigates these effects by replaying network activity on its own.
+
+### Web Server Variability
+
+Web servers have variable load and do not always respond with the same delay. Lower traffic sites with shared hosting infrastructure are typically more susceptible to this. applied throttling partially masks these effects by applying a minimum request latency in its network throttling. Simulated throttling is susceptible to this effect but the overall impact is usually low when compared to other network variability.
```
[the remark this answers is not in the document]
Comment (patrickhulce) [reply docs/variability.md]: sure sg

File: docs/variability.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+### Client Resource Contention
+
+Other applications running on the same machine while Lighthouse is running can cause contention for CPU, memory, and network resources. Malware, browser extensions, and anti-virus software have particularly strong impacts on web performance. Multi-tenant or parallel server environments can also suffer from these issues. applied throttling is susceptible to this issue. Simulated throttling partially mitigates this issue by replaying network activity on its own and capping CPU execution.
+
+### Browser Nondeterminism
+
+Browsers have inherent variability in their execution of tasks that impacts the way webpages are loaded. This is unavoidable for applied throttling as at the end of the day they are simply reporting whatever was observed by the browser. Simulated throttling is able to partially mitigate this effect by simulating execution on its own and only re-using task execution times from the browser in its estimate.
```
[the remark this answers is not in the document]
Comment (patrickhulce) [reply docs/variability.md]: done

File: docs/variability.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+### Page Nondeterminism
+
+Pages can contain logic that is nondeterministic that changes the way a user experiences a page, i.e. an A/B test that changes the layout and assets loaded or a different ad experience based on campaign progress. This is an intentional and irremovable source of variance. If the page changes in a way that hurts performance, Lighthouse should be able to identify this case. The only mitigation here is on the part of the site owner in ensuring that the exact same version of the page is being tested between different runs.
```
[the remark this answers is not in the document]
Comment (patrickhulce) [reply docs/variability.md]: yeah sg

File: docs/variability.md (hunk 1 above)
Comment (patrickhulce) [reply docs/variability.md]: this is referring to running lighthouse in multi-tenant server environments (like travis/AWS/etc) or multiple lighthouse/browser instances at once

File: docs/variability.md (hunk 2 above)
Comment (patrickhulce) [reply docs/variability.md]: yeah that sounds good, the docs I pulled these snippets from talked about this more at length so a reminder here is a good call 👍

**Pull request merged**